FirebaseDynamicLinks/Sources/FIRDynamicLinkNetworking.m

```diff
@@ -286,54 +270,40 @@ - (void)sendRequestWithBaseURLString:(NSString *)baseURL
-    }
-    NSString *matchMessage = nil;
-    NSError *parsingError = nil;
-    NSDictionary *parsedDynamicLinkParameters =
-        parserBlock(requestURLString, data, &matchMessage, &parsingError);
-
-    // If request was made with pasteboard contents, verify if we got a unique match. If we got
-    // a "none" match, we were unable to get a unique match or deduce using fingerprinting.
-    // In this case, resend requests to IPV4 and IPV6 endpoints for fingerprinting. b/79704203
-    if (requestBody[@"uniqueMatchLinkToCheck"] && parsedDynamicLinkParameters &&
-        (!parsedDynamicLinkParameters[kFIRDLParameterMatchType] ||
-         [parsedDynamicLinkParameters[kFIRDLParameterMatchType] isEqualToString:@"none"])) {
-      NSMutableDictionary *requestBodyMutable = [requestBody mutableCopy];
-      [requestBodyMutable removeObjectForKey:@"uniqueMatchLinkToCheck"];
-      NSMutableArray *baseURLs =
-          [@[ kIosPostInstallAttributionRestBaseUrlIPV4, kIosPostInstallAttributionRestBaseUrlIPV6 ]
-              mutableCopy];
-      if (parsedDynamicLinkParameters[kFIRDLParameterRequestIPVersion]) {
-        if ([parsedDynamicLinkParameters[kFIRDLParameterRequestIPVersion]
-                isEqualToString:@"IP_V4"]) {
-          [baseURLs removeObject:kIosPostInstallAttributionRestBaseUrlIPV4];
-        } else if ([parsedDynamicLinkParameters[kFIRDLParameterRequestIPVersion]
-                       isEqualToString:@"IP_V6"]) {
-          [baseURLs removeObject:kIosPostInstallAttributionRestBaseUrlIPV6];
+  FIRNetworkRequestCompletionHandler completeInvitationByDeviceCallback =
+      ^(NSData *data, NSError *error) {
+        if (error || !data) {
+          dispatch_async(dispatch_get_main_queue(), ^{
+            handler(nil, nil, error);
+          });
+          return;
         }
-      }
-
-      for (NSString *baseURL in baseURLs) {
-        [self sendRequestWithBaseURLString:baseURL
-                               requestBody:requestBodyMutable
-                              endpointPath:@"installAttribution"
-                               parserBlock:parserBlock
-                                completion:handler];
-      }
-    }
-    // We want to return out the result of the unique match check irrespective of success/failure as
-    // it is the first fingerprinting request as well.
-    dispatch_async(dispatch_get_main_queue(), ^{
-      handler(parsedDynamicLinkParameters, matchMessage, parsingError);
-    });
-  };
+        NSString *matchMessage = nil;
+        NSError *parsingError = nil;
+        NSDictionary *parsedDynamicLinkParameters =
+            parserBlock(requestURLString, data, &matchMessage, &parsingError);
+
+        // If request was made with pasteboard contents, verify if we got a unique match. If we got
+        // a "none" match, we were unable to get a unique match or deduce using fingerprinting.
+        // In this case, resend requests to IPV4 and IPV6 endpoints for fingerprinting. b/79704203
```

Comment:
I'm thinking that this won't be an issue anymore - we're using the exact same endpoint so I would assume that this logic would live on the server - if there's no unique match it'd try to fingerprint instead. I think it's relatively safe to make that assumption, remove the code and wait for confirmation from @eldhosembabu.